api/src/pcapi/core/users/email/update.py

```diff
@@ -104,39 +203,48 @@ def full_email_update_by_admin(user: models.User, email: str) -> None:
     repository.save(user, admin_update_event)
 
 
-def get_no_active_token_key(user: models.User) -> str:
-    return f"update_email_active_tokens_{user.id}"
+def get_token_key(user: models.User, token_type: TokenType) -> str:
+    match token_type:
+        case TokenType.CONFIRMATION:
+            return f"update_email_confirmation_token_{user.id}"
+        case TokenType.VALIDATION:
+            return f"update_email_validation_token_{user.id}"
+        case _:
+            raise ValueError(f"Invalid token type: {token_type}")
 
 
-def check_no_active_token_exists(user: models.User, expiration_date: datetime) -> None:
-    """
-    Use a dummy counter to find out whether the user already has an
-    active token.
+def token_exists(user: models.User, token_type: TokenType) -> bool:
+    """Check if there is an active confirmation token for the user"""
```

Comment:
(détail / doc)

```suggestion
    """Check if there is an active token (of the requested type) for the user"""
```

Cette fonction permet de vérifier la présence de différents types de token.